virttest/cartesian_config.py

```diff
@@ -538,10 +538,64 @@ def dump(self, indent, recurse=False):
 match_substitute = re.compile("\$\{(.+?)\}")
 
 
+def _drop_suffixes(d):
+    """
+    Merge suffixes for same var, or drop off unnecessary suffixes
+
+    This step returns a copy of a suffix flattened dictionary.
+    """
+    # dictionary `d_flat' is going to be the mutated copy of `d`
+    d_flat = d.copy()
+    for key in d:
+        if key in _reserved_keys:
+            continue
+
+        if not isinstance(key, tuple):
+            continue
+
+        try:
+            # This file was invoked through cmdline
+            options.skipdups
+            skipdups = options.skipdups
+        except NameError:
```

Comment:
Is `NameError` the exception thrown? In Python 3 at least it seems to throw AttributeError:

```python
>>> import optparse
>>> parser = optparse.OptionParser()
>>> options, args = parser.parse_args()
>>> options.foobar
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Values' object has no attribute 'foobar'
```

Reply:
The code was probably developed during the Python 2 era and never migrated. Will fix this too.